src/runtime/runtime.ts

```diff
@@ -894,7 +818,7 @@ class JoinNode implements Node {
     return affectedConstraints;
   }
 
-  applyCombination(input:Change, prefix:ID[], transaction:number, round:number, results:ID[][]) {
+  applyCombination(index:Index, input:Change, prefix:ID[], transaction:number, round:number, results:ID[][]) {
```

Comment:
whups, we need to be more disciplined and name globals in a way that they won't get picked up.